stablehlo/CMakeLists.txt

```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+add_subdirectory(compatibility)
```

Comment:
I moved the VHLO dialect to `stablehlo/dialect`. Unsure if transforms fit in with the dialect folder as well. The reason I chose a separate directory in the first place is that compatibility infrastructure feels like a feature of the StableHLO opset the same way that a reference interpreter is a feature of StableHLO, separate from the dialect.

Let's talk about this a bit more, I'm going to address the rest of the comments and leave this open for now. Currently this is 12 files that would be added to `stablehlo/dialect` that are all related to compatibility infrastructure. Also part of me wants to split `CompatibilityPasses` into separate files in the near future, since the relation between legalizations to/from VHLO and the VHLO-to-version conversions are conceptually different.

Reply:
Followed up offline. Directory structure will be:

```
stablehlo/
    dialect/
        Vhlo*
        Chlo*
        Stablehlo*
    transforms/
        LegalizeStablehloToVhlo*
        VhloToVersion*

```